addons/rook_secret_handler.go

```diff
@@ -197,6 +197,9 @@ func updateStorageCluster(secretName, storageClusterName, storageClusterNamespac
 	}
 
 	// Update secret name
+	if sc.Spec.Mirroring == nil {
+		sc.Spec.Mirroring = &ocsv1.MirroringSpec{}
+	}
```

Comment:
Same as above.

Reply:
This place is different from above, the immediate if condition is raising nil pointer exception 
if !utils.ContainsString(sc.Spec.Mirroring.PeerSecretNames, secretName)

so it has to be initialized before this line.